library_generation/test/generate_library_integration_test.sh

```diff
@@ -56,42 +58,49 @@ protobuf_version=$(get_version_from_WORKSPACE "protobuf-" WORKSPACE "-")
-# generate GAPIC client library
-popd
-popd
-echo "Generating library from ${proto_path}, to ${destination_path}..."
-"${library_generation_dir}"/generate_library.sh \
--p "${proto_path}" \
--d "${destination_path}" \
---gapic_generator_version "${gapic_generator_version}" \
---protobuf_version "${protobuf_version}" \
---grpc_version "${grpc_version}" \
---transport "${transport}" \
---rest_numeric_enums "${rest_numeric_enums}" \
---include_samples "${include_samples}"
-
-echo "Generate library finished."
-echo "Checking out googleapis-gen repository..."
-
-pushd "${output_folder}"
-sparse_clone "${googleapis_gen_url}" "${proto_path}"
+popd # googleapis
+popd # output_folder
 
-echo "Compare generation result..."
-RESULT=0
-diff -r "googleapis-gen/${proto_path}/${destination_path}" "${output_folder}/${destination_path}" -x "*gradle*" || RESULT=$?
+grep -v '^ *#' < "${proto_path_list}" | while IFS= read -r line; do
+  proto_path=$(echo "$line" | cut -d " " -f 1)
+  destination_path=$(echo "$line" | cut -d " " -f 2)
+  # parse GAPIC options from proto_path/BUILD.bazel
+  pushd "${output_folder}"
+  proto_build_file_path="${proto_path}/BUILD.bazel"
+  transport=$(get_transport_from_BUILD "${proto_build_file_path}")
+  rest_numeric_enums=$(get_rest_numeric_enums_from_BUILD "${proto_build_file_path}")
+  include_samples=$(get_include_samples_from_BUILD "${proto_build_file_path}")
+  popd # output_folder
+  echo "GAPIC options are transport=${transport}, rest_numeric_enums=${rest_numeric_enums}, include_samples=${include_samples}."
+  # generate GAPIC client library
+  echo "Generating library from ${proto_path}, to ${destination_path}..."
+  "${library_generation_dir}"/generate_library.sh \
+  -p "${proto_path}" \
+  -d "${destination_path}" \
+  --gapic_generator_version "${gapic_generator_version}" \
+  --protobuf_version "${protobuf_version}" \
+  --grpc_version "${grpc_version}" \
+  --transport "${transport}" \
+  --rest_numeric_enums "${rest_numeric_enums}" \
+  --include_samples "${include_samples}"
+  echo "Generate library finished."
+  echo "Checking out googleapis-gen repository..."
 
-if [ ${RESULT} == 0 ] ; then
-  echo "SUCCESS: Comparison finished, no difference is found."
-else
-  echo "FAILURE: Differences found."
-fi
+  echo "Compare generation result..."
+  pushd "${output_folder}"
+  sparse_clone "${googleapis_gen_url}" "${proto_path}/${destination_path}"
+  RESULT=0
+  # include gapic_metadata.json and package-info.java after
+  # resolving https://github.com/googleapis/sdk-platform-java/issues/1986
+  diff -r "googleapis-gen/${proto_path}/${destination_path}" "${output_folder}/${destination_path}" -x "*gradle*" -x "gapic_metadata.json" -x "package-info.java" || RESULT=$?
 
-popd # output_folder
+  if [ ${RESULT} == 0 ] ; then
+    echo "SUCCESS: Comparison finished, no difference is found."
+  else
+    echo "FAILURE: Differences found in proto path: ${proto_path}."
+    exit "${RESULT}"
```

Comment:
Would it make sense to build a list of failed libraries and then exit if the list is non emptyto have better insight on potential problems?

Reply:
I think the integration tests here is to mimic what happens in the real world, so we need to modify `generate_library.sh` if any difference exists.

We do have unit tests to verify failed cases (e.g., invalid versions).